### PR TITLE
Redefine 'Guides', merge old guides into API docs

### PR DESCRIPTION
**Pull request by rhyolight**
Better structure for docs. Converts "Guide" section into "Guides" with content that used to be in the Wiki. Moved previous "Guide" docs into the proper places inside the API docs.

### CODE REVIEW

**Comment by rhyolight** [comment]
CC @marionleborgne

File: docs/source/guides/swarming/algorithm.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+For sprint #0, all possible single fields are evaluated. For sprint #1 and subsequent sprints, only the top N (the default is 5, set via a configuration variable) fields from sprint #0 are considered. This keeps the swarm to a manageable size even if the dataset contains many, many fields.
+
+Each sprint contains one or more “mini-swarms” (MS). There is one MS for each field combination in the sprint. For each MS, we evaluate multiple models (typically dozens) and eventually settle in on the best model found for that particular field combination.  Within an MS, swarming will search over the parameter space and determine which model components to include in the model as well as the optimal values for each of the parameters in each component. The logic used within an MS is described below in the sections entitled “Choosing Scalar Parameter Values” and “Choosing Enumerated Parameter Values”. .
```
Comment (oxtopus) [inline docs/source/guides/swarming/algorithm.md]: Extra period at the end?

File: docs/source/guides/swarming/algorithm.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+When a user launches a swarm, they can set the swarm size to either “small”, “medium” or “large”. Currently, a “small” swarm creates just one particle per MS, a “medium” swarm creates 5 particles per MS, and a “large” swarm creates 15 particles per MS. Having more particles in an MS allows the logic to search the space more completely and can often yield a better performing model.
+
+## Choosing Enumerated Parameter Values
+This section describes the logic used within an MS to settle in on the best values for the enumerated parameters of a model. An example of an enumerate parameter is a Boolean parameter whose possible values are just true and false. Another example is a parameter that could take one of a list of possible values such as: [“scalar”, “delta”, “adaptive”].
+
+Some enumerated parameters are component parameters. For example, the SP’s “randomSP” parameter is a Boolean that controls whether or not the SP learns.
```
Comment (oxtopus) [inline docs/source/guides/swarming/algorithm.md]: is this bit about "randomSP" correct?

```
macbook-pro-2:nupic amarshall$ git grep randomSP
docs/source/guides/swarming/algorithm.md:Some enumerated parameters are component parameters. For example, the SP’s “randomSP” parameter is a Boolean that controls whether or not the SP learns.
```

Comment (rhyolight) [inline docs/source/guides/swarming/algorithm.md]: you're right, that param is gone now

**Comment by rhyolight** [comment]
@oxtopus Thanks for the review. Took all your suggestions.

**Pull request merged**